src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -252,12 +253,27 @@ internal struct NameDeclarationInfo
             {
                 result = IsFollowingTypeOrComma<VariableDeclarationSyntax>(token, semanticModel,
                      v => v.Type,
-                     v => v.Parent is LocalDeclarationStatementSyntax l ? l.Modifiers : default(SyntaxTokenList?),
+                     v => v.Parent is LocalDeclarationStatementSyntax l ? l.Modifiers :
+                          v.Parent is ForStatementSyntax ? default(SyntaxTokenList) : null as SyntaxTokenList?,
                      d => ImmutableArray.Create(SymbolKind.Local),
                      cancellationToken);
                 return result.Type != null;
             }
 
+            private static bool IsForEachVariableDeclaration(SyntaxToken token, SemanticModel semanticModel,
+                int position, CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                // This is parsed as ForEachVariableStatementSyntax:
+                // foreach (int $$
+                result = IsLastTokenOfType<CommonForEachStatementSyntax>(token, semanticModel,
+                    f => f is ForEachStatementSyntax forEachStatement ? forEachStatement.Type :
+                         ((ForEachVariableStatementSyntax)f).Variable,
```

Comment:
this formatting is not great.  consider either:

1. keeping it all on one line.
2. wrapping both ? and :
3. extracting out a helper method for this sort of thing.

Reply:
This is consistent with how this is formatted above:
```c#
                     v => v.Parent is LocalDeclarationStatementSyntax l ? l.Modifiers :
                          v.Parent is ForStatementSyntax ? default(SyntaxTokenList) : null as SyntaxTokenList?,
```

Reply:
I formatted it like this (sort of like a switch expression) because originally I had 2 is ?; checks but then removed the other one. I'll just put it back.